x-pack/plugins/cases/server/authorization/audit_logger.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OperationDetails } from '.';
+import { AuditLogger, EventCategory, EventOutcome } from '../../../security/server';
+
+enum AuthorizationResult {
+  Unauthorized = 'Unauthorized',
+  Authorized = 'Authorized',
+}
+
+export class AuthorizationAuditLogger {
+  private readonly auditLogger?: AuditLogger;
+
+  constructor(logger: AuditLogger | undefined) {
```

Comment:
Why it can be `undefined`?

Reply:
It's because we get the `AuditLogger` from the security plugin and the security plugin can be undefined (my guess is that'll happen when security is disabled).